test/new-e2e/processes/docker_test.go

```diff
@@ -0,0 +1,173 @@
+func TestDockerTestSuite(t *testing.T) {
+	suite.Run(t, new(DockerTestSuite))
+}
+
+func (s *DockerTestSuite) SetupSuite() {
+	var err error
+
+	s.ec2, err = NewEC2TestEnv("process-agent-docker-test")
+	require.NoError(s.T(), err)
+
+	waitForDocker(s.T(), s.ec2.sshClient)
+
+	configuration := common.NewConfiguration()
+	s.ddAPIClient = common.NewAPIClient(configuration)
+	s.procAPIClient = datadog.NewProcessesApi(s.ddAPIClient)
+}
+
+func (s *DockerTestSuite) TearDownSuite() {
+	s.ec2.Close()
+}
+
+func (s *DockerTestSuite) TearDownTest() {
+	killAndRemoveContainers(s.T(), s.ec2.sshClient)
+}
+
+func (s *DockerTestSuite) TestProcessAgentOnDocker() {
+	hostName := createHostName(s.T().Name())
+	hostTag := fmt.Sprintf("host:%s", hostName)
+
+	s.T().Logf("start the datadog agent with hostname: %s", hostName)
+	// run the agent container on the VM
+	stdout, err := clients.ExecuteCommand(s.ec2.sshClient, fmt.Sprintf("sudo docker run -d --cgroupns host"+
+		" --name %s"+
+		" -v /var/run/docker.sock:/var/run/docker.sock:ro"+
+		" -v /proc/:/host/proc/:ro"+
+		" -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro"+
+		" -v /dd/config/:/etc/datadog-agent/"+
+		" -e DD_PROCESS_AGENT_ENABLED=true"+
+		" -e DD_HOSTNAME=%s"+
+		" -e DD_API_KEY=%s %s", containerName, hostName, s.ec2.ddAPIKey, agentImage))
+	s.T().Log(stdout)
+	require.NoError(s.T(), err)
+
+	s.waitForProcessAgent()
+
+	// Start containers
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name spring-hello -p 8080:8080 -d springio/gs-spring-boot-docker")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name sleepy  -d busybox sleep 9999")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	// Check processes are
+	ctx := common.NewDefaultContext(context.Background())
+	assert.Eventually(s.T(), func() bool {
+		resp, _, err := s.procAPIClient.ListProcesses(ctx, *datadog.NewListProcessesOptionalParameters().
+			WithSearch("java").
+			WithTags(fmt.Sprintf("%s,container_name:spring-hello", hostTag)).
+			WithPageLimit(2),
+		)
+		if err != nil {
+			s.T().Logf("Error when calling `ProcessesApi.ListProcesses`go: %v\n", err)
+			return false
+		}
+		if procs, ok := resp.GetDataOk(); ok && procs != nil {
+			return len(*procs) == 1
+		}
+		return false
+	}, 2*time.Minute, 1*time.Second)
+
+	assert.Eventually(s.T(), func() bool {
+		resp, _, err := s.procAPIClient.ListProcesses(ctx, *datadog.NewListProcessesOptionalParameters().
+			WithTags(fmt.Sprintf("%s,container_name:sleepy", hostTag)).
+			WithPageLimit(2),
+		)
+		if err != nil {
+			s.T().Logf("Error when calling `ProcessesApi.ListProcesses`: %v\n", err)
+			return false
+		}
+		if procs, ok := resp.GetDataOk(); ok && procs != nil {
+			return len(*procs) == 1
+		}
+		return false
+	}, 2*time.Minute, 1*time.Second)
+
+}
+
+func (s *DockerTestSuite) waitForProcessAgent() {
+	var stdout string
+	require.Eventuallyf(s.T(), func() bool {
```

Comment:
I am guessing `require` here ant other `waitForX` steps is due to the fact we want to abort the test if we can't get to a running state?